sdk/src/Core/Internal/Emitters/UdpSegmentEmitter.cs

```diff
@@ -149,20 +151,34 @@ protected virtual void Dispose(bool disposing)
 
         private void SetEndPointOrDefault(string daemonAddress)
         {
-            IPEndPoint daemonEndPoint;
-
             if (string.IsNullOrEmpty(daemonAddress))
             {
-                 daemonEndPoint = new IPEndPoint(_defaultDaemonAddress, _defaultDaemonPort);
-                _logger.InfoFormat("Using default daemon address: {0}:{1}", daemonEndPoint.Address.ToString(), daemonEndPoint.Port);
+                EndPoint = new IPEndPoint(_defaultDaemonAddress, _defaultDaemonPort);
+                _logger.InfoFormat("Using default daemon address: {0}:{1}", EndPoint.Address.ToString(), EndPoint.Port);
             }
-            else if (!IPEndPointExtension.TryParse(daemonAddress, out daemonEndPoint))
+            else if (IPEndPointExtension.TryParse(daemonAddress, out var daemonEndPoint))
             {
-                daemonEndPoint = new IPEndPoint(_defaultDaemonAddress, _defaultDaemonPort);
-                _logger.InfoFormat("The given daemonAddress ({0}) is invalid, using default daemon address {1}:{2}.", daemonAddress, daemonEndPoint.Address.ToString(), daemonEndPoint.Port);
+                EndPoint = daemonEndPoint;
+            }
+            else if (HostEndpoint.TryParse(daemonAddress, out var hostEndpoint))
+            {
+                try
+                {
+                    EndPoint = hostEndpoint.GetIPEndPoint();
```

Comment:
Hi @emmekappa ,
Thank you for the PR.
Instead of getting a `IPEndPoint` instance for a hostname, should we just use `hostEndPoint `in the [_udpClient.Send()](https://github.com/aws/aws-xray-sdk-dotnet/blob/master/sdk/src/Core/Internal/Emitters/UdpSegmentEmitter.cs#L77).

We do have inbuilt method provided by .NET : [Link](https://msdn.microsoft.com/en-us/library/acf44a1a(v=vs.110).aspx) on UDP client object, which accepts hostname and port.

For example the `send()` call would look like : `_udpClient.Send(data, data.Length, hostEndPoint.Host, hostEndPoint.Port);`
This will  eliminate the need of DNS look up and rely on underlying implementation for IP endpoint resolver.

Please let me know your thoughts.

Reply:
Hi @emmekappa ,
Would you like to provide your thoughts on the comment. Also, have you got chance on DNS caching implementation as per the thread : aws/aws-xray-sdk-dotnet#19 ?

Best,
Yogi

Reply:
Hi @yogiraj07, sorry but I have had zero-time during the last weeks to work on this. 
We found too much issues by using this library and more in general the X-Ray service. I totally understand your points both on your feedback on my PR and the DNS caching issue but unfortunately I don't think I'll have time to work on this in a short period.

By the way I hope that my feedback could help you guys to improve the library, because I think that this could be a nice service for .NET developers.